File: applications/paper/transport_torus/mapc2m_torus.f
      subroutine mapc2m_torus(xc,yc,xp,yp,zp)
      implicit none

      double precision xc,yc,xp,yp,zp

      double precision alpha, beta
      common /torus_comm/ alpha, beta

      double precision pi, pi2
      common /compi/ pi, pi2

      double precision r1, R, theta, phi

      call map_comp2torus(xc,yc,theta,phi)      

      r1 = alpha*(1 + beta*sin(theta))
      R = 1 + r1*cos(phi)

      xp = R*cos(theta)
      yp = R*sin(theta)
      zp = r1*sin(phi)

      end

File: applications/paper/transport_torus/mapc2m_torus.m
function [xp,yp,zp] = mapc2m_torus2(xc,yc,alpha,beta,tr,pr)

if (nargin < 4)
    beta = 0;
end

pi2 = 2*pi;

theta = pi2*(tr(1) + (tr(2)-tr(1))*xc);
phi = pi2*(pr(1) + (pr(2) - pr(1))*yc);

r = alpha*(1 + beta*sin(theta));
R = 1 + r.*cos(phi);

xp = R.*cos(theta);
yp = R.*sin(theta);
zp = r.*sin(phi);

end